core/controller/src/main/scala/whisk/core/controller/WebActions.scala

```diff
@@ -440,21 +441,14 @@ trait WhiskWebActionsApi extends Directives with ValidateRequestSize with PostAc
     }
   }
 
-  /**
-   * Gets package from datastore.
-   * This method is factored out to allow mock testing.
-   */
-  protected def getPackage(pkgName: FullyQualifiedEntityName)(implicit transid: TransactionId): Future[WhiskPackage] = {
-    WhiskPackage.get(entityStore, pkgName.toDocId)
-  }
-
   /**
    * Gets action from datastore.
-   * This method is factored out to allow mock testing.
+   * if it is contained in the package, then resolve and merge parameters
+   * This method is factored out to allow mock testing
    */
-  protected def getAction(actionName: FullyQualifiedEntityName)(
+  protected def resolveActionAndMergeParameters(actionName: FullyQualifiedEntityName)(
```

Comment:
We should add a comment that while parameters are merged, annotations are not. This is important because the web annotation is on an action not an entire package. We should separately add a test for this invariant.

Reply:
need to update the docs here: https://github.com/apache/incubator-openwhisk/blob/master/docs/webactions.md#additional-features the precedence order would now also include package binding parameters.

Reply:
ok I'll update the docs soon.

Reply:
thank you - it might be worth adding a section in the docs about web actions for packages and bindings (taking in some of my comments below). what do you think?